test/snapd_test.dart

```diff
@@ -3375,25 +3373,25 @@ void main() {
         timestamp: DateTime(2000),
         snap: 'snap',
         interface: 'home',
-        constraints: const SnapdConstraints(
-          pathPattern: '/home/user/Downloads/*',
-          permissions: ['read'],
-        ),
-        outcome: SnapdRequestOutcome.allow,
-        lifespan: SnapdRequestLifespan.forever,
+        constraints: <String, dynamic>{
+          'pathPattern': '/home/user/Downloads/*',
+          'permissions': ['read'],
+          'outcome': SnapdRequestOutcome.allow.toString(),
```

Comment:
Since `constraints` is now an arbitrary JSON blob, would it be better to avoid using the `SnapdRequestOutcome` enum here? It's only being kept around for backwards compatibility right now and isn't technically part of the new API.

Reply:
good point: it _is_ part of the new API within the security center, but here we probably shouldn't be making any assumptions around the structure of the constraints blob.